### PR TITLE
fix(coppa): move to age input coppa

### LINKED ISSUE #3137: Move to age based COPPA
image

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes #3137

### CODE REVIEW

File: app/scripts/lib/auth-errors.js [hunk 1]
```diff
@@ -151,10 +151,6 @@ function (_, Errors) {
       errno: 1011,
       message: t('Valid email required')
     },
-    YEAR_OF_BIRTH_REQUIRED: {
```
Comment (shane-tomlinson) [inline app/scripts/lib/auth-errors.js]: We spoke at one point about how we were going to handle deprecated client-side errors. The resolution was to comment out the errors and leave them in the code. This is usually considered bad practice with version control, but is reasonable from a documentation perspective and ensuring somebody doesn't come along and try to fill in one of the newly empty numbers.

Comment (vladikoff) [inline app/scripts/lib/auth-errors.js]: sounds good!

File: tests/functional/firefox/same_account.js [hunk 2]
```diff
@@ -1,188 +0,0 @@
...
-], function (intern, registerSuite, require, restmail, TestHelpers) {
-  var config = intern.config;
-  var EMAIL_SERVER_ROOT = config.fxaEmailRoot;
-  var PASSWORD = 'password';
-  var TOO_YOUNG_YEAR = new Date().getFullYear() - 13;
-  var user;
-  var email;
-
-  registerSuite({
-    name: 'about:accounts signup',
```
Comment (shane-tomlinson) [inline tests/functional/firefox/same_account.js]: This suite is no longer needed?

File: tests/functional/lib/helpers.js [hunk 3]
```diff
@@ -438,11 +438,12 @@ define([
         .type(password)
       .end()
 
-      .findByCssSelector('#fxa-age-year')
+      .findByCssSelector('#age')
+      .type('4')
```
Comment (shane-tomlinson) [inline tests/functional/lib/helpers.js]: nit: indentation.

File: tests/functional/oauth_preverified_sign_up.js [hunk 4]
```diff
@@ -49,17 +48,18 @@ define([
           .type(PASSWORD)
         .end()
 
-        .findByCssSelector('#fxa-age-year')
-        .end()
-
-        .findById('fxa-' + (TOO_YOUNG_YEAR - 1))
-          .click()
+        .findByCssSelector('#age')
+        // XXX: Bug in Selenium 2.47.1, if Firefox is out of focus it will just type 1 number,
```
Comment (shane-tomlinson) [inline tests/functional/oauth_preverified_sign_up.js]: ugh. 

nit: indentation.

File: tests/functional/oauth_sign_up_verification_redirect.js [hunk 5]
```diff
@@ -39,7 +36,7 @@ define([
           return self.remote.get(require.toUrl(url + '&verification_redirect=always'));
         })
         .then(function () {
-          return FunctionalHelpers.fillOutSignUp(self, email, PASSWORD, OLD_ENOUGH_YEAR);
+          return FunctionalHelpers.fillOutSignUp(self, email, PASSWORD);
```
Comment (shane-tomlinson) [inline tests/functional/oauth_sign_up_verification_redirect.js]: I haven't mentioned, but I am really glad you simplified the function signature.

**Comment by ryanfeeley** [comment]
@vladikoff I need to UI review. What's the flag to get this to appear?

File: tests/functional/sign_up.js [hunk 6]
```diff
@@ -247,19 +225,11 @@ define([
...
-        .findById('fxa-month-' + monthToSelect)
-          .click()
-        .end()
-
-        .findById('fxa-day-' + dateToSelect)
-          .click()
+        .findByCssSelector('#age')
+        // XXX: Bug in Selenium 2.47.1, if Firefox is out of focus it will just type 1 number,
+        // split the type commands for each character to avoid issues with the test runner
+        .type('1')
```
Comment (shane-tomlinson) [inline tests/functional/sign_up.js]: nit: indentation.

File: tests/functional/sign_up.js [hunk 7]
```diff
@@ -215,26 +212,7 @@ define([
...
-      // provided for dayValue, the date will be set to the last day of the
-      // previous month.
-      now.setDate(now.getDate() + 1);
-      var monthToSelect = now.getMonth();
-      var dateToSelect = now.getDate();
-
+    'coppa does not allow sign up if younger than 13 years old': function () {
       return this.remote
         .get(require.toUrl(PAGE_URL))
         .findByCssSelector('form input.email')
```
Comment (shane-tomlinson) [inline tests/functional/sign_up.js]: Would `FunctionalHelpers.fillOutSignUp` with an optional age work here?

File: tests/functional/sign_up.js (hunk 6 above)
Comment (shane-tomlinson) [inline tests/functional/sign_up.js]: When you do the `.type`, do you have to do a `.click` first so the element receives the focus? I could swear that's why we do stuff like https://github.com/mozilla/fxa-content-server/pull/3138/files#diff-addb444c51a96421c45bfd482b36edd8R224

**Comment by vladikoff** [comment]
@ryanfeeley https://latest.dev.lcip.org/signup?forceExperiment=coppaView&forceExperimentGroup=treatment

File: tests/functional/firefox/same_account.js (hunk 2 above)
Comment (vladikoff) [inline tests/functional/firefox/same_account.js]: Yeah we don't use it

File: tests/functional/sign_up.js [hunk 8]
```diff
@@ -416,7 +341,6 @@ define([
 
   function testRepopulateFields(dest, header) {
     var self = this;
-    var year = OLD_ENOUGH_YEAR;
 
     return self.remote
```
Comment (shane-tomlinson) [inline tests/functional/sign_up.js]: Could `fillOutSignUp` with an optional age be used here as well?

**Comment by vladikoff** [comment]
image

**Comment by ryanfeeley** [comment]
👍

File: tests/functional/sign_up.js (hunk 6 above)
Comment (vladikoff) [inline tests/functional/sign_up.js]: Tried that trick, Sadly that does not help

**Comment by shane-tomlinson** [comment]
Testing now.

**Pull request merged**